homeassistant/components/solarlog/sensor.py

```diff
@@ -247,7 +247,11 @@ class SolarLogInverterSensorEntityDescription(SensorEntityDescription):
         native_unit_of_measurement=UnitOfPower.WATT,
         device_class=SensorDeviceClass.POWER,
         state_class=SensorStateClass.MEASUREMENT,
-        value_fn=lambda inverter: inverter.current_power,
+        value_fn=(
+            lambda inverter: None
+            if inverter.current_power is None
+            else inverter.current_power
+        ),
```

Comment:
But if `inverter.current_power is None` and you return `None`, isn't that the same as just returning `inverter.current_power`? Because if it is `None`, you return `None`, and if it is not `None`, it will return its value

Reply:
I think you are right. But why had we added this for consumption_year (L263 ff.)?

Reply:
No clue, haha

Reply:
Had a new thought on this and changed my mind. We added it, as the test goes for `inverter` and not for `inverter.current_power`. If `inverter` is `None`, then it cannot access `inverter.current_power` as no inverter data is available (for whatever reason). So I guess it makes sense to set the sensor values to `None`, showing them as unavailable.

Reply:
Can you please confirm, before I change it back again? ;-)

Reply:
```py
value_fn=(
            lambda inverter: None
            if inverter is not None
            else inverter.current_power
        ),
```
But then I would expect something like this, where we check if the inverter is not None